internal/shared/version/version.go

```diff
@@ -29,18 +47,20 @@ func init() {
 	for _, setting := range info.Settings {
 		switch setting.Key {
 		case "vcs.revision":
-			if gitCommit == "unknown" {
+			if isUnset(gitCommit) {
 				gitCommit = setting.Value
 			}
 		case "vcs.time":
-			commitDate = setting.Value
+			if isUnset(commitDate) {
+				commitDate = setting.Value
+			}
 		case "vcs.modified":
 			if v, ok := stateMap[setting.Value]; ok {
 				repoState = v
 			}
```

Comment:
```suggestion
			if v, ok := stateMap[setting.Value]; ok && isUnset(repoState) {
				repoState = v
			}
```

Shouldn't it be like?